homework-submission/src/store.js

```diff
@@ -0,0 +1,40 @@
+import {observable,action,computed,configure} from 'mobx';
+import stateLists from './listJson.json';
+configure({ enforceActions: true })
+class StoreTodo  {
+  @observable todos ={
+    stateLists,
+    description:'',
+    deadline:'',
+    showEdit:false
+  };
+@action onChange(index,event){
```

Comment:
I dont see any reason for not using onchange and ondelete or oncreate on mobx action. I have searched also a naming conventions of mobx which i ended up no written conventions for mobx action,if you have let me know to see it and believe.
every action or properties of a class has a very unique address which helps the computer to identify from the others. on the other way if i use such naming in the action the it will be easy for me to understand the name and its work and it gives me sense too.
any way i can accept your comment but let me see a prove of this on some documents that says what you are saying.

Reply:
Gide, is it not just about Mobx, it is about any frontend framework or any frontend logic. You can read about separation of concerns or MVC for example.

Frontend project is splitted up into two parts
1. for managing UI
2. for managing DATA

First one should have all stuff related to UI and managing components events .. (like onChange, onDelete)

2. Is more related to Data and Backend requests which is should have valid naming which relates to the request that you are making like (deleteTodo, createTodo ...).

This is coming out of experience during the years and working with different projects with different teams, if you like the way you name your actions .. then good it is up to you for sure ;)

I'll try to find a good article about his 👍

Reply:
I think Nouran already responded to this. Your React components deal with UI and responding to browser events. The store deals with managing data. By maintaining Separation of Concerns it is easier to deal with complexity (especially when your application grows).

BTW, asking your mentors for proof for their advice strikes me as inappropriate. Asking for clarification, by al means.

Reply:
if i give clear and understandable name to  a function ,doesnt mean i have mixed up uI and data.
as far as i keep their function is different and become more clear though.
Dont worry ,i have come through different programming languages, i understand the ui and data logic. 
and @remarcmij , who are u? I'm afraid talking people who  i dont know and does not have connection with before. is that normal?

Reply:
Hi Gide, I'm the balding, gray-haired  guy that you may have bumped in at HYF. I've been programming for the last 40+ years.

Reply:
but u ended up u dont have experience at all ,cos programming experience with out respecting otheres is just like you are good for nothing. no wonder i have 12 years of programming experiece, any way lets stop conversation as it can lead as to other situation, i dont need to confronted with any one. i hope this is the end of our comments  between me and you. bybyby good luck.